jenkins/crawler.go

```diff
@@ -56,30 +60,75 @@ func (c *Crawler) CrawlJenkins(buildURI string, buildID string) {
 
 	go c.updateState(buildID)
 	go c.crawlBuild(uri)
+	go c.logBuildLogs(buildID, uri)
+
 	c.processChannel <- buildID
 }
 
+func (c *Crawler) logBuildLogs(buildID string, uri *url.URL) {
+	for {
+		select {
+		case jlogs := <-c.logChannel:
+			c.log.Debug("got request to log the jenkins build logs")
+			for _, jlog := range jlogs {
+				c.printBuildLog(jlog, uri, buildID)
+			}
+		}
+	}
+}
+
+func (c *Crawler) printBuildLog(jlog *ale.Log, uri *url.URL, buildID string) {
+	jsonData, err := json.Marshal(jlog)
+	if err != nil {
+		c.log.WithError(err).Error("unable to parse log entry")
+	}
+	c.log.WithFields(logrus.Fields{
+		"uri":      uri.String(),
+		"build_id": buildID,
+	}).Info(string(jsonData))
+}
+
+func (c *Crawler) extractBuildLogs(jdata *ale.JenkinsData) []*ale.Log {
+	var jlogs []*ale.Log
+	for _, stage := range jdata.Stages {
+		if stage.SubStages != nil && len(stage.SubStages) > 0 {
+			for _, substage := range stage.SubStages {
+				jlogs = append(jlogs, substage.Logs...)
+			}
+		} else {
+			jlogs = append(jlogs, stage.Logs...)
+		}
+	}
+	return jlogs
+}
+
 func (c *Crawler) updateState(buildID string) {
 	for {
 		select {
 		case jdata := <-c.stateChannel:
-			logrus.Debug("got request to update the state")
+			c.log.Debug("got request to update the state")
 			if err := c.database.Put(jdata, buildID); err != nil {
-				logrus.WithField("build_id", buildID).WithError(err).Error("unable to add to database")
+				c.log.WithField("build_id", buildID).WithError(err).Error("unable to add to database")
 			}
-			logrus.WithField("build_id", buildID).Info("database updated")
+			c.log.WithField("build_id", buildID).Info("database updated")
 
 			if jdata.Status == "" || jdata.Status == "IN_PROGRESS" {
 				go func() {
-					logrus.Debug("sleeping for 5 seconds before requerying")
+					c.log.Debug("sleeping for 5 seconds before requerying")
 					time.Sleep(5 * time.Second)
 					c.processChannel <- buildID
 				}()
 			} else {
```

Comment:
I'm thinking of removing the `else` block here. WDYT? @alde @jeffersongirao 

![image](https://user-images.githubusercontent.com/4631744/63681438-15045b00-c7f6-11e9-8dde-01ea85db9173.png)